lib/constants/platforms.ts

```diff
@@ -19,4 +19,5 @@ export const GITLAB_API_USING_HOST_TYPES = [
   'gitlab-releases',
   'gitlab-tags',
   'gitlab-packages',
+  'gitlab-changelog',
```

Comment:
@viceice are you sure you like this idea?

Reply:
I have looked for alternative solutions to solve the problem. 

The problem is that it is currently impossible to apply hostRules to calls made by the changelog modules. This due to the fact that they use the platform identifier as hostType, which cannot have any rules applied to them since the automatically generated host rule for the platform is always applied last, overriding the authentication set by any previous rule. 

Most datasources apply their datasource id to their Http client which adds it as hostType, making it possible to set hostRules for specific datasources. That is also what gitlab-tags and gitlab-releases datasources do.  The hostRules also have a fallback mechanism for sources that use Gitlab or Github as source.  This makes it possible for gitlab-tags and gitlab-releases to have specific rules configured for private repositories and otherwise fallback to the platform configuration. The changelog http clients,(which are called after datasource lookups) however always use the platform identifier as hostType, which as previously mentioned cannot be overridden.

I can think of only the 3 options to make hostRules configurable for changelogs so that they don't fail when datasources are configured with their own authentication which is not applied to subsequent changelog lookups.

1. Moving automatically generated platform rule to the beginning, making it possible to override automatically generated rule for call using the platform identifier. Making it possible to override the configured renovate platform token. However there is no ways to differentiate between preset, platform and changelog calls. 
   - Which might also break configurations where there are silently ignored hostRules for the platformId hostType.

2. In the case of the Gitlab changelog retrieval, it does 3 types of calls, tag lookup, checking for existence of changelog md(tree lookup) and looking up releases.
    - tags lookup could use the same hostType that gitlab-tags datsource uses
    - release lookup could use the same hostType that gitlab-releases datsource uses
    - the tree api lookup call is done also done in the platform module with platform identifier as hostType, but that was the problem, so new identifier needs to be introduced, since both tags and releases types calls don't make sense.
    
 3. Treat the changelog module just like a gitlab-releases or gitlab-tags datasource with its own identifier used as hostType, making it possible to differentiate from other platform calls and apply rules to them, and adding it to the fallback list. (The solution above, which I personally like best, but I am biased.)